spacy/language.py

```diff
@@ -922,7 +1223,70 @@ def _link_components(self):
                     if hasattr(proc2, "model"):
                         proc1.find_listeners(proc2.model)
 
-    def to_disk(self, path, exclude=tuple()):
+    @classmethod
+    def from_config(
+        cls,
+        config: Union[Dict[str, Any], Config] = {},
+        disable: Iterable[str] = tuple(),
+        overrides: Dict[str, Any] = {},
+        auto_fill: bool = True,
+        validate: bool = True,
+    ) -> "Language":
+        """Create the nlp object from a loaded config. Will set up the tokenizer
+        and language data, add pipeline components etc. If no config is provided,
+        the default config of the given language is used.
+        """
+        if auto_fill:
+            config = util.deep_merge_configs(config, cls.default_config)
+        if "nlp" not in config:
+            raise ValueError(Errors.E985.format(config=config))
+        nlp_config = config["nlp"]
+        config_lang = nlp_config["lang"]
+        if cls.lang is not None and config_lang is not None and config_lang != cls.lang:
+            raise ValueError(
+                Errors.E958.format(
+                    bad_lang_code=nlp_config["lang"],
+                    lang_code=cls.lang,
+                    lang=util.get_object_name(cls),
+                )
+            )
+        nlp_config["lang"] = cls.lang
+        # This isn't very elegant, but we remove the [pipeline] block here to prevent
+        # it from getting resolved (causes problems because we expect to pass in
+        # the nlp and name args for each component). If we're auto-filling, we're
+        # using the nlp.config with all defaults.
+        config = Config().from_str(config.to_str())  # deepcopy config!
```

Comment:
So the obvious mistake people will make as you've pointed out before, is people will pass actual instances in the config, like e.g. as a `Vocab`instance or a model or something. What will the failure look like here if they do that?

Also, can we just implement a `Config.copy` method in Thinc?

Reply:
Ah, I should add my `is_json_serializable` check here and do a proper deep copy. Forgot to add that here. See error `E961` for the message we raise.